### PR TITLE
CI setup for running tests in drone

### LINKED ISSUE #101: Drone setup
Move from Travis to Drone

### PR DESCRIPTION
**Pull request by dpakach**
# drone CI setup

fixes #101

- Implement style check for owncloud coding standard
- Add make targets for running tests
- Add `.drone.yml` file for drone configurations

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -54,5 +54,13 @@ nbproject
...
+/tests/output/
 
 build
+
+.php_cs.cache
+
+# Composer
+composer.phar
+vendor/
+vendor-bin/**/composer.lock
```
Comment (phil-davis) [inline .gitignore]: Also add:
```
vendor-bin/**/vendor
```
like I just did in https://github.com/owncloud/files_antivirus/pull/273/files#r235873391

Some day we will have all the pieces of the puzzle in one place.

File: .php_cs.dist [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+
+$config = new OC\CodingStandard\Config();
+
+$config
+	->setUsingCache(true)
+	->getFinder()
+	->exclude('l10n')
+	->exclude('vendor')
+	->exclude('vendor-bin')
+	->exclude('c3.php')
```
Comment (phil-davis) [inline .php_cs.dist]:
```
	->notPath('/^c3.php/')
```
see the current way this is done in ``password_policy``

File: Makefile [hunk 3]
```diff
@@ -26,6 +26,10 @@ endif
 endif
 endif
 
+# bin file definitions
+PHPUNIT=php -d zend.enable_gc=0  vendor/bin/phpunit
```
Comment (phil-davis) [inline Makefile]: See https://github.com/owncloud/files_antivirus/pull/273 and https://github.com/owncloud/password_policy/pull/166

Now we are running ``phpunit`` from where it is installed in core.

File: Makefile [hunk 4]
```diff
@@ -26,6 +26,10 @@ endif
...
+PHPUNITDBG=phpdbg -qrr -d memory_limit=4096M -d zend.enable_gc=0 "./vendor/bin/phpunit"
```
Comment (phil-davis) [inline Makefile]: Same here - change location of phpunit

File: composer.json [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+  "name": "owncloud/announcementcenter",
+  "config" : {
+    "platform": {
+      "php": "5.6.37"
+    }
+  },
+  "require": {
+  },
+  "require-dev": {
+    "phpunit/phpunit": "^5.7",
```
Comment (phil-davis) [inline composer.json]: this should no longer be needed

File: composer.json [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    "jakub-onderka/php-parallel-lint": "^1.0",
```
Comment (phil-davis) [inline composer.json]: Are we keeping any lint?
@DeepDiver1975 @patrickjahns do we remove all evidence of PHP-lint-related testing?

File: phpunit.xml [hunk 7]
```diff
@@ -1,24 +1,24 @@
...
 		 strict="true"
 		 verbose="true"
 		 timeoutForSmallTests="900"
 		 timeoutForMediumTests="900"
 		 timeoutForLargeTests="900"
 		>
-	<testsuite name='ownCloud - Announcement Center App Tests'>
-		<directory suffix='Test.php'>.</directory>
+	<testsuite name='unit'>
+		<directory suffix='Test.php'>./tests</directory>
```
Comment (phil-davis) [inline phpunit.xml]: The existing unit tests should be moved into ``tests/unit`` folder.
I guess we can try and do all this in a single PR!

Comment (dpakach) [reply phpunit.xml]: @phil-davis I tried moving them in unit folder but doing that breaks the unit tests. I think all the files namespace need changing we should try doing that in another PR

File: composer.json (hunk 6 above)
Comment (phil-davis) [reply composer.json]: This has been sorted out today. We remove lint checks and add owncloud-coding-standard for PHP 5.6 - that gets us a PHP 5.6 syntax check.

Comment (dpakach) [reply composer.json]: removed php-lint

**Pull request merged**